### PR TITLE
[JENKINS-54678] Fix for compressed log support

### PR DESCRIPTION
**Pull request by ArturHarasimiuk**
See [JENKINS-54678](https://issues.jenkins-ci.org/browse/JENKINS-54678).

Signed-off-by: Artur Harasimiuk <artur.harasimiuk@intel.com>

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java [hunk 1]
```diff
@@ -167,8 +167,17 @@
...
-            // Similar to Run.getLogFile, but not supporting gzip:
-            return FileLogStorage.forFile(new File(b.getRootDir(), "log"));
+
+            File rawFile = new File(b.getRootDir(), "log");
+            if(rawFile.isFile()) {
+                return FileLogStorage.forFile(rawFile);
+            }
+
+            File gzFile = new File(b.getRootDir(), "log.gz");
+            if (gzFile.isFile()) {
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java]:
```suggestion
            if (gzFile.isFile()) { // Are we looking at a completed build with compressed builds logs?
```

File: src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java [hunk 2]
```diff
@@ -167,8 +167,17 @@
                     return storage;
                 }
             }
-            // Similar to Run.getLogFile, but not supporting gzip:
-            return FileLogStorage.forFile(new File(b.getRootDir(), "log"));
+
+            File rawFile = new File(b.getRootDir(), "log");
+            if(rawFile.isFile()) {
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java]:
```suggestion
            if (rawFile.isFile()) { // Are we looking at a complete build with uncompressed logs or an in-progress build that has already written to the logs?
```

File: src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java [hunk 3]
```diff
@@ -167,8 +167,17 @@
...
+                return FileLogStorage.forFile(rawFile);
+            }
+
+            File gzFile = new File(b.getRootDir(), "log.gz");
+            if (gzFile.isFile()) {
+                return FileLogStorage.forFile(gzFile);
+            }
+            return FileLogStorage.forFile(rawFile);
```
Comment (dwnusbaum) [inline src/main/java/org/jenkinsci/plugins/workflow/log/LogStorage.java]:
```suggestion
            return FileLogStorage.forFile(rawFile); // No log file exists (probably a new build), use the standard uncompressed log file
```

**Comment by bitwiseman** [comment]
Tests never implemented.  Closing.  Happy to reopen if updated.

**Pull request closed without merging**